securedrop_client/gui/widgets.py

```diff
@@ -1152,19 +1152,24 @@ def update(self):
             )
             raise
 
-    def set_snippet(self, source, uuid=None, content=None):
+    def set_snippet(self, source, uuid=None, content=None) -> None:
         """
         Update the preview snippet only if the new message is for the
         referenced source and there's a source collection. If a uuid and
         content are passed then use these, otherwise default to whatever the
         latest item in the conversation might be.
         """
-        if source == self.source.uuid and self.source.collection:
+        if source != self.source.uuid:
```

Comment:
this could crash if the source was deleted and we try to access the uuid

Reply:
I added some padding here; please take another look. :slightly_smiling_face: